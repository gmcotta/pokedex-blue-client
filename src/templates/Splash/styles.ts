import styled, { css, keyframes, DefaultTheme } from 'styled-components';

export const Container = styled.main`
  height: 100%;
  position: relative;
  overflow: hidden;
`;

type AnimationProps = {
  animationTriggered: boolean;
};

export const Device = styled.section<AnimationProps>`
  ${({ theme, animationTriggered }) => css`
    width: 100%;
    height: 200%;
    background-color: ${theme.colors.pokedex.primary};
    position: absolute;
    bottom: 0;
    ${animationTriggered &&
    css`
      bottom: -100%;
      transition: bottom 500ms 300ms ease-in-out;
    `}
  `}
`;

export const UpperScreen = styled.section`
  ${({ theme }) => css`
    height: 50%;
    background-color: ${theme.colors.pokedex.primary};
    display: flex;
    justify-content: center;
    align-items: center;
    z-index: 10;
    position: relative;
  `}
`;

const turnOnDisplay = (theme: DefaultTheme) => keyframes`
  from {
    background-color: ${theme.colors.pokedex.screenOff};
    border-radius: 0.8rem;
    border: min(5vh, 3.2rem) solid ${theme.colors.pokedex.secondary};
    width: 95%;
    height: 85%;
  }
  to {
    background-color: ${theme.colors.pokedex.screenOn};
      border: 0;
      border-radius: 0;
      width: 100%;
      height: 100%;
  }
`;

export const Display = styled.div<AnimationProps>`
  ${({ theme, animationTriggered }) => css`
    background-color: ${theme.colors.pokedex.screenOff};
    border-radius: 0.8rem;
    border: min(5vh, 3.2rem) solid ${theme.colors.pokedex.secondary};
    width: 95%;
    height: 85%;
    ${animationTriggered &&
    css`
      animation: ${turnOnDisplay(theme)} 500ms 1000ms ease-in-out;
      animation-fill-mode: forwards;
    `}
  `}
`;

export const BottomDetail = styled.div<AnimationProps>`
  ${({ theme, animationTriggered }) => css`
    background-color: ${theme.colors.pokedex.secondary};
    width: 100%;
    height: min(5vh, 3.2rem);
    position: absolute;
    bottom: 0;
    ${animationTriggered &&
    css`
      opacity: 0;
      transition: opacity 200ms 1000ms linear;
    `}
  `}
`;

export const LowerScreen = styled.section`
  ${({ theme }) => css`
    height: 50%;
    background-color: ${theme.colors.pokedex.primary};
    display: flex;
    align-items: center;
    justify-content: center;
    position: relative;
  `}
`;

export const PokeballDecoration = styled.div`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    width: min(102vw, 80rem);
    height: min(102vw, 80rem);
    background-color: transparent;
    border-radius: 50%;
    border: min(5vw, 3.2rem) solid ${theme.colors.pokedex.secondary};
    /* margin: 1.6rem; */

    &::after {
      content: '';
      width: min(5vw, 3.2rem);
      height: 120vh;
      background-color: ${theme.colors.pokedex.secondary};
      position: absolute;
    }
  `}
`;

export const OutterBall = styled.div`
  ${({ theme }) => css`
    width: min(80vw, 40rem);
    height: min(80vw, 40rem);
    background-color: ${theme.colors.pokedex.secondary};
    border-radius: 50%;
    position: relative;
    display: flex;
    justify-content: center;
    align-items: center;
    margin: 1.6rem;

    &::after {
      content: ' ';
      width: min(80vw, 40rem);
      height: min(5vw, 3.2rem);
      background-color: ${theme.colors.pokedex.primary};
      position: absolute;
    }
  `}
`;

export const InnerBall = styled.div`
  ${({ theme }) => css`
    box-sizing: content-box;
    width: min(20vw, 10rem);
    height: min(20vw, 10rem);
    background-color: ${theme.colors.pokedex.secondary};
    border-radius: 50%;
    border: min(5vw, 3.2rem) solid ${theme.colors.pokedex.primary};
    z-index: 2;
  `}
`;

type RectangleButtonProps = {
  position?: 'topLeft' | 'topRight';
};

export const RectangleButton = styled.div<RectangleButtonProps>`
  ${({ theme, position = 'topLeft' }) => css`
    height: min(3vw, 3.2rem);
    width: min(10vw, 9.6rem);
    background-color: ${theme.colors.grayLight};
    box-shadow: ${theme.effects.shadow.insideWhite},
      ${theme.effects.shadow.outsideBlack};
    position: absolute;
    top: clamp(2.4rem, 1vw, 4.8rem);
    ${position === 'topLeft' &&
    css`
      left: clamp(2.4rem, 1vw, 4.8rem);
    `}
    ${position === 'topRight' &&
    css`
      right: clamp(2.4rem, 1vw, 4.8rem);
    `}
  `}
`;

export const RightButtonsWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  position: absolute;
  bottom: clamp(2.4rem, 1vw, 4.8rem);
  right: clamp(2.4rem, 1vw, 4.8rem);
  transform: rotateZ(15deg);
`;

export const SmallCircleButton = styled.div`
  ${({ theme }) => css`
    height: min(4vw, 3.6rem);
    width: min(4vw, 3.6rem);
    border-radius: 50%;
    background-color: ${theme.colors.grayLight};
    box-shadow: ${theme.effects.shadow.insideWhite},
      ${theme.effects.shadow.outsideBlack};
  `}
`;

export const BigCircleButton = styled.div`
  ${({ theme }) => css`
    height: min(8vw, 8rem);
    width: min(8vw, 8rem);
    border-radius: 50%;
    background-color: ${theme.colors.grayLight};
    box-shadow: ${theme.effects.shadow.insideWhite},
      ${theme.effects.shadow.outsideBlack};
    margin: min(4vw, 2.4rem) 0;
  `}
`;

export const DPadWrapper = styled.div`
  ${() => css`
    position: absolute;
    left: clamp(2.4rem, 1vw, 4.8rem);
    bottom: clamp(3.6rem, 1vw, 9.6rem);
    display: grid;
    grid-template-areas: '. top .' 'left center right' '. bottom .';
  `}
`;

export const DPadButton = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.grayLight};
    height: min(6vw, 4rem);
    width: min(6vw, 4rem);
    grid-area: center;
  `}
`;

export const DPadLeft = styled(DPadButton)`
  box-shadow: -4px 1px 4px rgba(0, 0, 0, 0.25),
    -4px -4px 4px rgba(0, 0, 0, 0.25);
  grid-area: left;
`;

export const DPadTop = styled(DPadButton)`
  box-shadow: -3px -4px 4px rgba(0, 0, 0, 0.25),
    -4px 0px 4px rgba(0, 0, 0, 0.25);
  grid-area: top;
`;

export const DPadRight = styled(DPadButton)`
  box-shadow: inset -4px -4px 4px #ffffff;
  grid-area: right;
`;

export const DPadBottom = styled(DPadButton)`
  box-shadow: inset -4px -4px 4px #ffffff;
  grid-area: bottom;
`;

export const Overlay = styled.div<AnimationProps>`
  ${({ animationTriggered }) => css`
    position: fixed;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    background-color: rgba(0, 0, 0, 0.3);
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    z-index: 10;
    ${animationTriggered &&
    css`
      opacity: 0;
      transition: opacity 300ms ease-in-out;
    `}
  `}
`;

export const TitleLogo = styled.img`
  width: 90%;
`;

export const Subtitle = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xl};
    color: #c4dced;

    @media (max-width: ${theme.breakpoints.md}) {
      font-size: ${theme.font.sizes.lg};
    }

    @media (max-width: ${theme.breakpoints.sm}) {
      font-size: ${theme.font.sizes.md};
    }
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background: linear-gradient(180deg, #83b8e9 25%, #2390f4 100%);
    font-size: ${theme.font.sizes.sm};
    color: ${theme.colors.grayLight};
    height: 4.8rem;
    padding: 0.8rem 1.6rem;
    border-radius: 0.8rem;
    margin-top: 2.4rem;
    border: 0;
    border-bottom: 4px solid #2d41f0;

    &:hover {
      filter: brightness(0.8);

      cursor: pointer;
      transition: filter 200ms ease-in;
    }

    &:active {
      box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.25);
    }
  `}
`;
