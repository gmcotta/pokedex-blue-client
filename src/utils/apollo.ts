import {
  ApolloClient,
  HttpLink,
  InMemoryCache,
  NormalizedCacheObject,
} from '@apollo/client';
import { useMemo } from 'react';

let apolloClient: ApolloClient<NormalizedCacheObject>;

const isOnServer = typeof window === 'undefined';

function createApolloClient() {
  return new ApolloClient({
    link: new HttpLink({ uri: 'http://localhost:1337/graphql' }),
    cache: new InMemoryCache(),
    ssrMode: isOnServer,
  });
}

export function initializeApollo(initialState = {}) {
  const apolloClientGlobal = apolloClient ?? createApolloClient();
  if (initialState) {
    apolloClientGlobal.cache.restore(initialState);
  }
  if (isOnServer) return apolloClientGlobal;
  apolloClient = apolloClient ?? apolloClientGlobal;
  return apolloClient;
}

export function useApollo(initialState = {}) {
  const store = useMemo(() => initializeApollo(initialState), [initialState]);
  return store;
}
