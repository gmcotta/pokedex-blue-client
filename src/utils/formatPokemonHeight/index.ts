function formatPokemonHeight(height: number) {
  return `${height / 10} m`;
}

export default formatPokemonHeight;
