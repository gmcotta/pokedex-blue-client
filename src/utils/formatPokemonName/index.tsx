function formatPokemonName(id: number, name: string): string {
  return `#${id.toString().padStart(3, '0')} - ${name}`;
}

export default formatPokemonName;
