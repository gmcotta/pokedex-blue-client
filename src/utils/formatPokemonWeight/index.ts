function formatPokemonWeight(weight: number) {
  return `${weight / 10} kg`;
}

export default formatPokemonWeight;
