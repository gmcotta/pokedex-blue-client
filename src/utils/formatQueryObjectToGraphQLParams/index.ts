import { ParsedUrlQuery } from 'querystring';

export function formatQueryObjectToGraphQLParams(queryObj: ParsedUrlQuery) {
  let finalFilter = {};
  if (queryObj.types) {
    const typeFilter = [];
    const typesSelected = (queryObj.types as string).split(',');
    for (const type of typesSelected) {
      typeFilter.push({
        name: {
          eq: type,
        },
      });
    }
    finalFilter = {
      types: {
        or: typeFilter,
      },
    };
  }
  return finalFilter;
}
