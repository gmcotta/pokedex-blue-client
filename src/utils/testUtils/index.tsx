import { ReactElement } from 'react';
import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';

import theme from '@/styles/theme';

export const renderWithTheme = (ui: ReactElement): RenderResult =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>);
