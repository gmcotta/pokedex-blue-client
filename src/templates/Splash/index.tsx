import Pokeball from 'components/Pokedex/Pokeball';
import * as S from './styles';

const SplashTemplate = () => {
  return (
    <S.Wrapper>
      <S.UpperScreen>
        <S.Display />
        <S.BottomDetail />
      </S.UpperScreen>
      <S.LowerScreen>
        <S.PokeballDecoration>
          <Pokeball />
        </S.PokeballDecoration>
        <S.RectangleButton position="topLeft" />
        <S.RectangleButton position="topRight" />
        <S.DPadWrapper>
          <S.DPadButton />
          <S.DPadTop />
          <S.DPadRight />
          <S.DPadBottom />
          <S.DPadLeft />
        </S.DPadWrapper>
        <S.RightButtonsWrapper>
          <S.SmallCircleButton />
          <S.BigCircleButton />
          <S.SmallCircleButton />
        </S.RightButtonsWrapper>
      </S.LowerScreen>
    </S.Wrapper>
  );
};

export default SplashTemplate;
