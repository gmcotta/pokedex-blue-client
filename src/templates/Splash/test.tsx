import { screen, waitFor, fireEvent } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { act } from '@testing-library/react-hooks';

import { renderWithTheme } from '@/utils/testUtils';

import Splash from '.';

// eslint-disable-next-line @typescript-eslint/no-var-requires
const useRouter = jest.spyOn(require('next/router'), 'useRouter');
const push = jest.fn();
const prefetch = jest.fn(() => Promise.resolve(true));
useRouter.mockImplementation(() => ({
  push,
  prefetch,
  query: '',
  asPath: '',
  route: '/',
}));

describe('<Splash />', () => {
  it('should render the component', () => {
    const { container } = renderWithTheme(<Splash />);
    expect(screen.getByAltText(/pokémon - título/i)).toBeInTheDocument();
    expect(
      screen.getByRole('heading', {
        name: /pokédex - react version/i,
      })
    ).toBeInTheDocument();
    expect(
      screen.getByRole('button', {
        name: /iniciar/i,
      })
    ).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should redirect to Pokémon List page', async () => {
    renderWithTheme(<Splash />);
    act(() => {
      userEvent.click(
        screen.getByRole('button', {
          name: /iniciar/i,
        })
      );
    });
    fireEvent.animationEnd(screen.getByRole('figure'));
    await waitFor(() => {
      expect(push).toHaveBeenCalledWith('/pokemon-list');
    });
  });
});
