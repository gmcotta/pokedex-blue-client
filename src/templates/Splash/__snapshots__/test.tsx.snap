// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Splash /> should render the component 1`] = `
.c0 {
  height: 100%;
  position: relative;
  overflow: hidden;
}

.c5 {
  width: 100%;
  height: 200%;
  background-color: #F64E4E;
  position: absolute;
  bottom: 0;
}

.c6 {
  height: 50%;
  background-color: #F64E4E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  z-index: 10;
  position: relative;
}

.c7 {
  background-color: #376473;
  border-radius: 0.8rem;
  border: min(5vh,3.2rem) solid #B03333;
  width: 95%;
  height: 85%;
}

.c8 {
  background-color: #B03333;
  width: 100%;
  height: min(5vh,3.2rem);
  position: absolute;
  bottom: 0;
}

.c9 {
  height: 50%;
  background-color: #F64E4E;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  position: relative;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: min(102vw,80rem);
  height: min(102vw,80rem);
  background-color: transparent;
  border-radius: 50%;
  border: min(5vw,3.2rem) solid #B03333;
}

.c10::after {
  content: '';
  width: min(5vw,3.2rem);
  height: 120vh;
  background-color: #B03333;
  position: absolute;
}

.c11 {
  width: min(80vw,40rem);
  height: min(80vw,40rem);
  background-color: #B03333;
  border-radius: 50%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 1.6rem;
}

.c11::after {
  content: ' ';
  width: min(80vw,40rem);
  height: min(5vw,3.2rem);
  background-color: #F64E4E;
  position: absolute;
}

.c12 {
  box-sizing: content-box;
  width: min(20vw,10rem);
  height: min(20vw,10rem);
  background-color: #B03333;
  border-radius: 50%;
  border: min(5vw,3.2rem) solid #F64E4E;
  z-index: 2;
}

.c13 {
  height: min(3vw,3.2rem);
  width: min(10vw,9.6rem);
  background-color: #EDEAEA;
  box-shadow: inset -4px -4px 4px #FFFFFF,-4px -4px 4px rgba(0,0,0,0.25);
  position: absolute;
  top: clamp(2.4rem,1vw,4.8rem);
  left: clamp(2.4rem,1vw,4.8rem);
}

.c14 {
  height: min(3vw,3.2rem);
  width: min(10vw,9.6rem);
  background-color: #EDEAEA;
  box-shadow: inset -4px -4px 4px #FFFFFF,-4px -4px 4px rgba(0,0,0,0.25);
  position: absolute;
  top: clamp(2.4rem,1vw,4.8rem);
  right: clamp(2.4rem,1vw,4.8rem);
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  bottom: clamp(2.4rem,1vw,4.8rem);
  right: clamp(2.4rem,1vw,4.8rem);
  -webkit-transform: rotateZ(15deg);
  -ms-transform: rotateZ(15deg);
  transform: rotateZ(15deg);
}

.c22 {
  height: min(4vw,3.6rem);
  width: min(4vw,3.6rem);
  border-radius: 50%;
  background-color: #EDEAEA;
  box-shadow: inset -4px -4px 4px #FFFFFF,-4px -4px 4px rgba(0,0,0,0.25);
}

.c23 {
  height: min(8vw,8rem);
  width: min(8vw,8rem);
  border-radius: 50%;
  background-color: #EDEAEA;
  box-shadow: inset -4px -4px 4px #FFFFFF,-4px -4px 4px rgba(0,0,0,0.25);
  margin: min(4vw,2.4rem) 0;
}

.c15 {
  position: absolute;
  left: clamp(2.4rem,1vw,4.8rem);
  bottom: clamp(3.6rem,1vw,9.6rem);
  display: grid;
  grid-template-areas: '. top .' 'left center right' '. bottom .';
}

.c16 {
  background-color: #EDEAEA;
  height: min(6vw,4rem);
  width: min(6vw,4rem);
  grid-area: center;
}

.c20 {
  box-shadow: -4px 1px 4px rgba(0,0,0,0.25),-4px -4px 4px rgba(0,0,0,0.25);
  grid-area: left;
}

.c17 {
  box-shadow: -3px -4px 4px rgba(0,0,0,0.25),-4px 0px 4px rgba(0,0,0,0.25);
  grid-area: top;
}

.c18 {
  box-shadow: inset -4px -4px 4px #ffffff;
  grid-area: right;
}

.c19 {
  box-shadow: inset -4px -4px 4px #ffffff;
  grid-area: bottom;
}

.c1 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: rgba(0,0,0,0.3);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  z-index: 10;
}

.c2 {
  width: 90%;
}

.c3 {
  font-size: 4.8rem;
  color: #c4dced;
}

.c4 {
  background: linear-gradient(180deg,#83b8e9 25%,#2390f4 100%);
  font-size: 1.6rem;
  color: #EDEAEA;
  height: 4.8rem;
  padding: 0.8rem 1.6rem;
  border-radius: 0.8rem;
  margin-top: 2.4rem;
  border: 0;
  border-bottom: 4px solid #2d41f0;
}

.c4:hover {
  -webkit-filter: brightness(0.8);
  filter: brightness(0.8);
  cursor: pointer;
  -webkit-transition: filter 200ms ease-in;
  transition: filter 200ms ease-in;
}

.c4:active {
  box-shadow: 0px 4px 4px rgba(0,0,0,0.25);
}

@media (max-width:768px) {
  .c3 {
    font-size: 3.2rem;
  }
}

@media (max-width:375px) {
  .c3 {
    font-size: 2.4rem;
  }
}

<main
  class="c0"
>
  <div
    class="c1"
  >
    <img
      alt="Pokémon - título"
      class="c2"
      src="/images/pokemon-title-logo.png"
    />
    <h2
      class="c3"
    >
      Pokédex - React Version
    </h2>
    <button
      class="c4"
    >
      Iniciar
    </button>
  </div>
  <section
    class="c5"
  >
    <section
      class="c6"
    >
      <div
        class="c7"
        role="figure"
      />
      <div
        class="c8"
      />
    </section>
    <section
      class="c9"
    >
      <div
        class="c10"
      >
        <div
          class="c11"
        >
          <div
            class="c12"
          />
        </div>
      </div>
      <div
        class="c13"
      />
      <div
        class="c14"
      />
      <div
        class="c15"
      >
        <div
          class="c16"
        />
        <div
          class="c16 c17"
        />
        <div
          class="c16 c18"
        />
        <div
          class="c16 c19"
        />
        <div
          class="c16 c20"
        />
      </div>
      <div
        class="c21"
      >
        <div
          class="c22"
        />
        <div
          class="c23"
        />
        <div
          class="c22"
        />
      </div>
    </section>
  </section>
</main>
`;
